onnxscript/function_libs/torch_aten/param_manipulation.py

```diff
@@ -0,0 +1,171 @@
+"""Function for manipulating input parameters of an Op or a OnnxFunction."""
+from __future__ import annotations
+
+import collections
+import dataclasses
+from typing import Any, List, OrderedDict, Sequence, Tuple
+
+import onnx
+
+from onnxscript import values
+
+# A special value to indicate that the default value is not specified
+_EmptyDefault = object()
+
+
+@dataclasses.dataclass(frozen=True)
+class ParamSchema:
+    """A schema for a parameter of an Op or a OnnxFunction.
+
+    Attributes:
+        name: The name of the parameter.
+        type: The type of the parameter.
+        default: The default value of the parameter.
+        is_input: Whether the parameter is an ONNX input.
+    """
+
+    name: str
+    type: Any = None  # Op input does not have a type, for now
+    default: Any = _EmptyDefault
+    is_input: bool = True
+
+    def __repr__(self) -> str:
+        param_kind = "INPUT" if self.is_input else "ATTRIBUTE"
+        text = f"{self.name}<{param_kind}>: {self.type}"
+        if self.default is not _EmptyDefault:
+            text += f" = {self.default}"
+        return text
+
+    @property
+    def is_attribute(self) -> bool:
+        """Returns True if the parameter is an ONNX attribute."""
+        return not self.is_input
+
+
+def extract_param_schema_from_function(onnx_func: values.OnnxFunction) -> List[ParamSchema]:
```

Comment:
The current `OnnxFunction` IR loses the positional-order information between inputs and attributes (since we didn't care about it then given our assumptions that all inputs come first). We can't reconstruct it subsequently.

Perhaps we should consider updating `OnnxFunction` IR to store a `List[ParamSchema]` in the first place ... just thinking out aloud, haven't read the rest of the PR yet.

Reply:
I like this idea! I was going to propose exposing ParamSchema somewhere in the function and op as the next thing to do so we have a common interface as #320